docs/extending_deptrac.md

```diff
@@ -0,0 +1,96 @@
+# Extending Deptrac
+
+Deptrac defines its extension point by providing set of **contract** classes
+that you can depend on for your implementation. The classes can be found in
+the `src/Contract` directory and are covered by
+the [backwards compatibility policy](bc_policy.md).
+
+There are several ways you can extend Deptrac:
+
+- [Output formatters](#Output Formatters) - To change how the output of
+  dependency analysis is formatted. This might be useful to integrate Deptrac
+  into your CI/CD pipeline.
+- [Analyser Events](#Analyser Events) - Decide whether a dependency is
+  Uncovered, Allowed, Violation or SkippedViolation.
+- [Layer collectors](#Layer collectors) - Add custom deptrac collector
+
+> **Note**
+> In examples where FQCN is not specified, the base
+> namespace `Qossmic\Deptrac\Contract\` is omitted for readability.
+
+## Output Formatters
+
+Creating an output formatter requires only to create a new class implementing
+the `OutputFormatter\OutputFormatterInterface` and
+register it
+in your `deptrac.yaml` file like this:
+
+```yaml
+services:
+  - class: <Your FQCN>
+    autowire: true
+    tags:
+      - output_formatter
+```
+
+And you are done. You can call your formatter by using the `-f` or `--formatter`
+CLI flag with the name you defined
+in `OutputFormatter\OutputFormatterInterface::getName()`.
+
+## Analyser Events
+
+The deptrac analyser creates a `Analyser\ProcessEvent` for each dependency it
+finds. You
+can listen for this event and yourself decide how to handle it depending on the
+details of the dependency.
+
+This might be for example useful when you want to ignore a specific group of
+violations (in testing code for example). First you create a class to process
```

Comment:
```suggestion
This might be useful when you want to ignore a specific group of
violations in testing-related code for example. First you create a class to process
```